### PR TITLE
test: fix test_combined_cloud_config_json

### PR DESCRIPTION
**Pull request by aciba90**
## Proposed Commit Message

```
test: fix test_combined_cloud_config_json

Features are series dependent, make the test aware of that.
The test now distiguishes between devel and in_place cloud-init pakages
which would carry quilt patches modifying features.

Co-authored-by: Chad Smith <chad.smith@canonical.com>
Co-authored-by: James Falcon <james.falcon@canonical.com>
```

## Test Steps

```
export CLOUD_INIT_CLOUD_INIT_SOURCE=PROPOSED
export CLOUD_INIT_PLATFORM=lxd_container
export CLOUD_INIT_OS_IMAGE=focal


tox -e integration-tests -- --pdb tests/integration_tests/modules/test_combined.py::TestCombined::test_combined_cloud_config_json
```

## Checklist

- [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/index.html)
- [ ] I have updated or added any [unit tests](https://cloudinit.readthedocs.io/en/latest/development/testing.html) accordingly
- [ ] I have updated or added any [documentation](https://cloudinit.readthedocs.io/en/latest/development/contribute_docs.html) accordingly

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: tests/integration_tests/modules/test_combined.py [hunk 1]
```diff
@@ -323,7 +329,12 @@ def test_combined_cloud_config_json(
             "/run/cloud-init/combined-cloud-config.json"
         )
         data = json.loads(combined_json)
-        assert data["features"] == get_features()
+        expected_features = get_features()
+        if CURRENT_RELEASE in (FOCAL, JAMMY, MANTIC):
```
Comment (TheRealFalcon) [inline tests/integration_tests/modules/test_combined.py:333]: Nit: You can do `if CURRENT_RELEASE < NOBLE:` here, but it looks like test is still failing.

**Review by blackboxsw — APPROVED** [review]
Approved with the following suggested change.

**Review by blackboxsw — CHANGES_REQUESTED** [review]
Given that our github CI is running against a package built for bionic, we're seeing CI failures now because we aren't building a package anymore that is aware of these features.py. Should be be bumping our `RELEASE: focal` as well?

**Comment by TheRealFalcon** [comment]
> package built for bionic

Doh! Yes...we should update that and should've when Bionic went EOL. I think the oldest supported LTS makes the most sense for CI.

**Comment by blackboxsw** [comment]
> > package built for bionic
> 
> Doh! Yes...we should update that and should've when Bionic went EOL. I think the oldest supported LTS makes the most sense for CI.

I added the following PR to bump to focal, since it needs a separate commit anyway. But beyond this I think our CI will still be unhappy because we are using packages/bddeb in .github/workflows/integration.yaml which doesn't have the quilt patches we actually carry on a given distribution series. I don't think we want to duplicate all the feature flag patching we do in downstream ubuntu distributions in packages/bddeb. So this may require a bit more smarts in this integration-test to understand what package we are actually testing, not just the ubuntu release

File: tests/integration_tests/modules/test_combined.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (blackboxsw) [reply tests/integration_tests/modules/test_combined.py:333]: For CI, which uses packages/bddeb and creates a named deb version containing `bddeb` I think we should just assert we see see the expected get_features() from the current repo.

```suggestion
        pkg_ver = client.execute("cloud-init -v").split()[-1]
        if "bddeb" not in pkg_ver and CURRENT_RELEASE < NOBLE:
```

**Comment by aciba90** [comment]
> I added the following PR. https://github.com/canonical/cloud-init/pull/4871 which bumps our CI package builder to focal since it needs a separate commit anyway.

Thanks for that!

> But beyond this issue, I think our CI will still be unhappy because we are using packages/bddeb in .github/workflows/integration.yaml which doesn't have the quilt patches we actually carry on a given distribution series. I don't think we want to duplicate all the feature flag patching we do in downstream ubuntu distributions in packages/bddeb.

> So this may require a bit more smarts in this integration-test to understand what package we are actually testing, not just the ubuntu release.

What should we do with this PR/test then? The features assertion does not add much value but friction, I believe, especially if we cannot check for stable release features.

File: tests/integration_tests/modules/test_combined.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (aciba90) [reply tests/integration_tests/modules/test_combined.py:333]: That could work, I think. We will have to take care about this check still working in the future, but works.

**Comment by aciba90** [comment]
I missed this comment: https://github.com/canonical/cloud-init/pull/4868#discussion_r1488263520

I went ahead and implement that suggestion, thanks!

**Review by blackboxsw — APPROVED** [review]
LGTM.

**Pull request merged**